### PR TITLE
Include version nicknames

### LINKED ISSUE #8: Return version nicknames?
Could this package also return version nicknames, either optionally in `r_versions()` or in a new function?

From R 2.15.1 there seems to be a special file with the info e.g. https://svn.r-project.org/R/tags/R-2-15-1/VERSION-NICK (but it's not always present :thinking: )

### PR DESCRIPTION
**Pull request by gaborcsardi**
Closes #8.

### CODE REVIEW

**Comment by maelle** [comment]
Oh cool I was starting to look at how to implement this! :eyes:

File: R/nicks.R [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    "3.3.0" = "Supposedly Educational", "3.3.1" = "Bug in Your Hair",
+    "3.3.2" = "Sincere Pumpkin Patch", "3.3.3" = "Another Canoe",
+    "3.4.0" = "You Stupid Darkness", "3.4.1" = "Single Candle",
+    "3.4.2" = "Short Summer", "3.4.3" = "Kite-Eating Tree",
+    "3.4.4" = "Someone to Lean On", "3.5.0" = "Joy in Playing",
+    "3.5.1" = "Feather Spray", "3.5.2" = "Eggshell Igloo",
+    "3.5.3" = "Great Truth")
+
+get_nicknames <- function(vers = r_versions(dots = TRUE)$version) {
+  dash_vers <- gsub(".", "-", vers, fixed = TRUE)
```
Comment (maelle) [inline R/nicks.R:48]: why do this rather than calling `r_versions(dots = FALSE)`?

File: R/nicks.R [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  first_nick <- match("2-15-0", dash_vers)
+  if (is.na(first_nick)) first_nick <- 1L
```
Comment (maelle) [inline R/nicks.R:50]: why would it be NA?

File: R/nicks.R [hunk 3]
```diff
@@ -0,0 +1,93 @@
+
+nicks <-
```
Comment (maelle) [inline R/nicks.R]: Should nicks be a function returning the nicks to not have to declare a global variable?

**Comment by maelle** [comment]
LGTM! Very cool! :smile_cat:

File: R/nicks.R (hunk 1 above)
Comment (gaborcsardi) [reply R/nicks.R:48]: because I forgot what that did....

File: R/nicks.R (hunk 2 above)
Comment (gaborcsardi) [reply R/nicks.R:50]: because 2-15-0 is not the requested versions. that's the case when the stored values are outdated.

File: R/nicks.R (hunk 3 above)
Comment (gaborcsardi) [reply R/nicks.R]: does not matter much in this case

Comment (gaborcsardi) [reply R/nicks.R]: although to test the on-demand nick update, a function is better bc it can be mocked easier

File: R/nicks.R (hunk 1 above)
Comment (gaborcsardi) [reply R/nicks.R:48]: otoh, for on-demand it might be called with dots, so....

File: R/nicks.R (hunk 3 above)
Comment (maelle) [reply R/nicks.R]: I meant a function returning this named vector, so apparently it would have no advantage :-)

**Comment by gaborcsardi** [comment]
Thanks!

**Pull request merged**